src/screens/BaseScreen/BaseScreen.jsx

```diff
@@ -1,33 +1,118 @@
-import { StyleSheet, Text, TouchableOpacity, View } from 'react-native';
+import { StyleSheet, Text, TouchableOpacity, View, Image } from 'react-native';
+import { useNavigation } from '@react-navigation/native';
 import PropTypes from 'prop-types';
 import React from 'react';
+import { Feather, AntDesign } from '@expo/vector-icons';
 
 import gStyle from '../../constants/gStyle';
 import BaseBottomTabNavigatorLayout from '../../layouts/BaseBottomTabNavigatorLayout';
 import colors from '../../constants/colors';
+import sharedStyles from '../../constants/sharedStyles';
+import FloatingActionButton from '../../components/FloatingActionButton';
+import CreateMsgIcon from '../../assets/icons/CreateMsgIcon';
+import LogoTextLinkItem from '../../components/LogoTextLinkItem';
+
+const mockChannels = ['General', 'random', 'Sphere Cube', 'BayBo'];
 
 const styles = StyleSheet.create({
-  container: {
-    alignItems: 'center',
+  headerContainer: {
+    flexDirection: 'row',
+    marginTop: 20,
+    width: '100%',
+    height: 150,
+    marginBottom: 10,
+  },
+  headerTextContainer: {
+    marginLeft: 20,
+  },
+  headerText: {
+    ...sharedStyles.textWhiteBold20,
+    marginBottom: 10,
+  },
+  image: {
+    width: 150,
+    height: 150,
+    borderRadius: 8,
+  },
+  scoreBox: {
+    width: 80,
     flex: 1,
+    backgroundColor: colors.white10,
+    borderRadius: 15,
     justifyContent: 'center',
+    alignItems: 'center',
+    marginTop: 10,
+  },
+  categoryContainer: {
+    justifyContent: 'space-between',
+    flexDirection: 'row',
+    alignItems: 'center',
+  },
+  categoryText: {
+    ...sharedStyles.textWhiteBold12,
+    marginVertical: 10,
   },
   text: {
     ...gStyle.textLarsBold18,
     color: colors.white,
   },
 });
 
-export default function BaseScreen({ currentBaseId, goToChat }) {
+export default function BaseScreen({ currentBaseId, baseEntities, userEntities, goToChat }) {
+  const navigation = useNavigation();
+  const base = baseEntities[currentBaseId];
   return (
     <BaseBottomTabNavigatorLayout>
-      <View style={styles.container}>
-        <Text style={[styles.text]}>BaseScreen</Text>
-        <Text style={[styles.text]}>{currentBaseId}</Text>
-        <TouchableOpacity style={{ padding: 20 }} onPress={() => goToChat()}>
-          <Text style={gStyle.listText}>Chats</Text>
+      <View style={styles.headerContainer}>
```

Comment:
i would separate this file out into multiple components, there are several sections that you could spit out just,

for non reusuable components (these are built just for this component) just put them into a components folder under BaseScreen